uniffi_bindgen/src/interface/types/codetype.rs

```diff
@@ -0,0 +1,210 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::interface::types::dispatch::*;
+use crate::interface::FFIType;
+
+type_dispatch! {
+    /// Type behavior for generated code
+    ///
+    /// This trait handles the language-agnostic behavior.  We also create a language-specific
+    /// version in the bindings/scaffolding code.
+    pub trait NewCodeType {
```

Comment:
What does the "new" in `NewCodeType` signify? I wasn't able to parse the name into a sensible mental model of what it's for in a way that makes use of "new".

Reply:
"New" just means "not the current one".  If we decide to go with this approach then the plan would be to update the swift code to use it as well, then rename `NewCodeType` -> `CodeType`.